go-controller/pkg/ovn/controller/apbroute/master_controller.go

```diff
@@ -429,13 +478,27 @@ func (c *ExternalGatewayMasterController) syncNamespace(namespace *v1.Namespace)
 }
 
 func (c *ExternalGatewayMasterController) onPodAdd(obj interface{}) {
-	c.podQueue.Add(obj)
+	pod := obj.(*v1.Pod)
```

Comment:
i think without ok on these you will still panic

Reply:
Updated for pod, namespace and policy to add check in add and update use cases.